lib/3scale/backend/listener.rb

```diff
@@ -576,6 +582,14 @@ def check_transactions_validity(transactions)
         end
       end
 
+      # In the past, the log field in a transaction could also include
+      # "response" and "request". Those fields are not used anymore, but some
+      # callers are still sending them. We want to filter them to avoid storing
+      # them in the job queues, decoding them, etc. unnecessarily.
+      def log_without_unused_attrs(log)
+        log.select { |k| k == 'code' }
```

Comment:
Using `log.select!` or `log.reject!` avoids duplicating the whole hash (note that they don't return the same as `select`/`reject`)

Reply:
Yeah, but the problem with `select!` is that it returns nil when it doesn't change the hash.

Reply:
Why is that a problem in this case?

Reply:
Changed :+1: